libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -76,9 +105,10 @@ struct CopyTranslate: public yul::ASTCopier
 		auto const& reference = m_references.at(&_identifier);
```

Comment:
Wouldn't it be cleaner to do all the references-related handling in the same function?

Reply:
I'd need to get one parameter in and about three out, not sure the end result will be cleaner.. (bool, reference, varDcl)

Reply:
I meant why not do all of it inside `yul::Expression operator()(yul::Identifier const& _identifier)`?

Reply:
I can't remember anymore why, but I do remember that using `yul::Expression operator()(yul::Identifier const& _identifier)` for everything didn't work. Something about the way the ASTCopier ist structured. If I wouldn't use `translate` it would miss references. 

And I can't do the slot / offset resolving in `translate()` because I need to return a literal (= expression).